detox/android/detox/src/full/java/com/wix/detox/reactnative/ReactNativeLoadingMonitor.kt

```diff
@@ -79,51 +76,34 @@ open class ReactNativeLoadingMonitor(
             }
         }
 
-        return rnInstanceManager.currentReactContext
+        return rnApplication.getCurrentReactContextSafe()
     }
-}
 
-private interface DummyListenerIdentifier
-
-/**
- * This baby bridges over RN's breaking change introduced in version 0.68:
- * `ReactInstanceManager$ReactInstanceEventListener` was extracted onto a separate interface, having
- * `ReactInstanceManager.{add|remove}ReactInstanceEventLister()` changing their signature to use it, accordingly.
- *
- * This made us resort to a solution based on dynamic proxies, because - depending on RN's version (at runtime), we
- * need to dynamically decide what interface we "extend" (or actually shadow) via the proxy.
- *
- * @see RNDiff https://github.com/facebook/react-native/compare/v0.67.4..v0.68.0#diff-2f01f0cd7ff8c9ea58f12ef0eff5fa8250cad144dd5490598d80d8e9e743458aR1009
- * @see DynamicProxies https://docs.oracle.com/javase/8/docs/technotes/guides/reflection/proxy.html
- */
-private fun subscribeAsyncRNContextHandler(rnInstanceManager: ReactInstanceManager, onReactContextInitialized: () -> Any) {
-    val listenerClass = resolveListenerClass()
-    val proxyInterfaces = arrayOf(
-        listenerClass,
-        DummyListenerIdentifier::class.java // In order to be able to implement equals()
-    )
-    val listener = Proxy.newProxyInstance(listenerClass.classLoader, proxyInterfaces) { listener, method, args ->
-        Log.d(LOG_TAG, "Listener-proxy method called: ${method.name}")
-
-        val result = when (method.name) {
-            "onReactContextInitialized" -> {
-                Log.i(LOG_TAG, "Got new RN-context async'ly through listener")
-                Reflect.on(rnInstanceManager).call("removeReactInstanceEventListener", listener)
-                onReactContextInitialized()
-            }
-            "equals" -> {
-                val candidate = args[0]
-                candidate is DummyListenerIdentifier
-            }
-            else -> Any()
+    private fun isReactNativeLoaded(reactContext: ReactContext?) =
+        reactContext != null && reactContext !== previousReactContext && reactContext.hasActiveReactInstance()
+
+    private fun subscribeAsyncRNContextHandler(onReactContextInitialized: () -> Any) {
+        val isFabric = isFabricEnabled()
+        if (isFabric) {
+            // We do a casting for supporting RN 0.75 and above
+            val host = rnApplication.reactHost as ReactHostImpl?
+            host?.addReactInstanceEventListener(object : ReactInstanceEventListener {
```

Comment:
`host?` is an issue... If we reach this subscribe() flow and host is null, we'd stay stuck forever

Reply:
I see but here the code implicitly suggests that adding the listener is just "nice-to-have", maybe `host!!` would make it more explicit